### PR TITLE
De-recommend checking full label consistency for custom collectors.

### PR DESCRIPTION
**Pull request by brian-brazil**
Signed-off-by: Brian Brazil <brian.brazil@robustperception.io>

@beorn7

### CODE REVIEW

File: content/docs/instrumenting/writing_clientlibs.md [hunk 1]
```diff
@@ -250,8 +250,9 @@ Client libraries MUST NOT under any circumstances allow users to have different
 label names for the same metric for Gauge/Counter/Summary/Histogram or any
 other Collector offered by the library.
 
-If your client library does validation of metrics at collect time, it MAY also
-verify this for custom Collectors.
+While metrics from custom collectors generally should also have consistent label
+names, as there are rare valid use cases where this is okay client libraries
```
Comment (RichiH) [inline content/docs/instrumenting/writing_clientlibs.md]: That sentence is more complex than it should be.

Comment (brian-brazil) [reply content/docs/instrumenting/writing_clientlibs.md]: What would you suggest?

Comment (RichiH) [reply content/docs/instrumenting/writing_clientlibs.md]: As a general rule, metrics from custom collectors generally should also have consistent label names. As there are still rare but valid use cases where this is not the case, client libraries should not verify this. An example of such a rare case would be XXX

Last sentence optional, but I can't think of any example off-hand, so including one seems useful to me.

Comment (brian-brazil) [reply content/docs/instrumenting/writing_clientlibs.md]: Adjusted

Comment (RichiH) [reply content/docs/instrumenting/writing_clientlibs.md]: Where? GH only shows me one commit; the initial one.

Comment (brian-brazil) [reply content/docs/instrumenting/writing_clientlibs.md]: Forgot to push it

**Comment by RichiH** [comment]
LGTM.

@beorn7 ?

**Comment by beorn7** [comment]
:+1:

**Pull request merged**